Code from Ash

johnash90 opened the pull request:
Upload of code to merge crash records with road segment information to receive comments from team

johnash90 committed:
Code from Ash

Upload of code to merge crash records with road segment information to receive comments from team

charleyjoo commented on crashDataAnalysisTools/merge_crash_and_segment.py:
```diff
@@ -0,0 +1,68 @@
+
+def merge_crash_and_segment(crashes, roadSegments):
+    '''
+    Inputs: - a csv file containing individual crash records for
```
Maybe the function input should be two string file names instead of two csv files?

charleyjoo commented on crashDataAnalysisTools/merge_crash_and_segment.py:
```diff
@@ -0,0 +1,68 @@
+    '''
+
+    # construct the filepaths for the csv files
+    crashFilePath = os.getcwd() + '/' + crashes + '.csv'
```
Lowercase letters are preferred for instant variable names based on PEP8.

charleyjoo commented on crashDataAnalysisTools/merge_crash_and_segment.py:
```diff
@@ -0,0 +1,68 @@
+
+    # query to join count total number of crashes occuring on
+    # each road segment based on route number and milepost
+    qryGetCrashCount = '''
```
Maybe some more aggregation statistics of crashes can be added?

johnash90 replied:
This is a good idea, I will look into this for the project!

charleyjoo commented on crashDataAnalysisTools/merge_crash_and_segment.py:
```diff
@@ -0,0 +1,68 @@
+    segmentCrashCount = pd.read_sql(qryGetCrashCount, con=db)
+
+    # merge the crash counts into the road segment data frame
+    roadSegmentsAndCrashCounts = pd.merge(roadSeg,
```
This step can be integrated into the SQL code by selecting all attributes from the roadSegments table.

charleyjoo commented on crashDataAnalysisTools/merge_crash_and_segment.py:
```diff
@@ -0,0 +1,68 @@
+    qryGetCrashCount = '''
+        SELECT RoadSegments.begmp, RoadSegments.endmp, RoadSegments.road_inv,
+        COUNT(Crashes.caseno) as CrashCount
+        FROM Crashes, RoadSegments
```
Maybe having two letters denoting the tables (e.g., c and r)? Cuz they appear multiple times below.

johnash90 committed:
Update merge_crash_and_segment.py based on comments

Per Wenbo's comments:
-Updated description of inputs from 'csv' to 'string file name'
-For the comments about sql code, since the code works for now, I will look into improving its readability for the final project
-Variable names updated to lower case with underscores

Per Hongyuan's comments:
-Indentation level of sql query text was adjusted to be more clear and valid per pep8
-Description of arguments changed from csv to file names
-Indicated that file name should not contain 'csv'
